packages/pyright-internal/src/analyzer/checker.ts

```diff
@@ -745,6 +762,8 @@ export class Checker extends ParseTreeWalker {
 
         this._scopedNodes.push(node);
 
+        // It visits all child nodes, but it changes process ordering from pre-order to post-order
+        // by doing its own walking.
         return false;
```

Comment:
there are a few visit methods that return false and does its own walking. I know some of them do that to avoid visit name but some of them, not sure why they are doing it. I tried change a few to true and see what tests are failing, but didn't try every cases. 

at some point, it would be nice if we can write it down the reason why it doesn't walk some of its child nodes. 

even if I find tests failing, and open the test files (since test itself just check error count but doesn't say what error), it just show the error without much explanation on why it fails with the walking child nodes.